test/node/session-resend-event-on-connection-drop.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const test = require('tap');
+
+const cp = require('child_process');
+const path = require('path');
+
+const jstp = require('../..');
+
+let client = getClient();
+
+const iface = 'iface';
+const event = 'testEvent';
+const ifaces = {
+  iface: {
+    method: (connection, callback) => {
+      client.send(['getSession']);
+      callback(null);
+    },
+  },
+};
+
+const application = new jstp.Application('testApp', ifaces);
+const serverConfig = { applications: [application] };
+const server = jstp.net.createServer(serverConfig);
+
+client.on('message', ([message, ...args]) => {
+  switch (message) {
+    case 'error':
+      test.fail('must not encounter an error');
+      break;
+    case 'session':
+      session(...args);
+      break;
+  }
+});
+
+server.listen(0, () => {
+  client.send(['connect', server.address().port]);
+});
+
+function getClient() {
+  return cp.fork(
+    path.join(__dirname, '../utils/session/client'),
+    [],
```

Comment:
You can omit the `args` argument, according to the [docs](https://nodejs.org/dist/latest-v6.x/docs/api/child_process.html#child_process_child_process_fork_modulepath_args_options) it is optional.